src/main/kotlin/app/revanced/patcher/util/xml/LazyXMLInputSource.kt

```diff
@@ -0,0 +1,12 @@
+package app.revanced.patcher.util.xml
+
+import com.reandroid.apk.xmlencoder.EncodeMaterials
+import com.reandroid.apk.xmlencoder.XMLEncodeSource
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.source.XMLDocumentSource
+
+internal class LazyXMLInputSource(name: String, val document: XMLDocument, materials: EncodeMaterials) : XMLEncodeSource(materials, XMLDocumentSource(name, document)) {
+    fun encode() {
+        resXmlBlock
+    }
+}
```

Comment:
Looks like unfinished code.

Reply:
It is not unfinished. In java, this would be `this.getResXmlBlock()`, which [does have side effects](https://github.com/REAndroid/ARSCLib/blob/de4a6feb245e25f17f9cce04c2f99505183e8e26/src/main/java/com/reandroid/apk/xmlencoder/XMLEncodeSource.java#L60-L74).

Reply:
The class has now been fully documented.